add Bash trap framework basics, refactored stack trace

stevekuznetsov opened the pull request:
This will begin the series of pulls that hopefully will bring in the trap framework. 

@Miciah please (re-)review the Bash bits
@smarterclayton please review the last commit

smarterclayton commented on hack/build-cross.sh:
```diff
@@ -29,7 +29,8 @@ fi
 OS_BUILD_PLATFORMS=("${platforms[@]}")
-os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+# Create a sub-shell so that we don't pollute the outer environment
+( os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}" )
```
Why not make build_binaries itself run its work in a subshell, rather than propagating up and out?

stevekuznetsov replied:
My question was, also, if we have a work-around for the `$GOBIN` thing for cross-compiles, why not use that work-around for everything and never leak a changed `$GOBIN`, thereby not needing the subshell at all?

smarterclayton commented on hack/build-go.sh:
```diff
@@ -10,7 +10,11 @@ STARTTIME=$(date +%s)
 source "${OS_ROOT}/hack/common.sh"
 source "${OS_ROOT}/hack/util.sh"
-os::log::install_errexit
+source "${OS_ROOT}/hack/lib/util/trap.sh"
```
Why do I have to import so many things?

stevekuznetsov replied:
1. not everyone is going to need everything
2. compartmentalizing our support functions helps people understand them
3. our "lets put everything in `common.sh`" approach failed as soon as the file got so long that multiple versions of the same function started showing up with different implementations.

Miciah replied:
Would it be weird to `source "${OS_ROOT}/hack/lib/util/trap.sh"` in `hack/lib/log/stacktrace.sh` and make `os::log::stacktrace::install` invoke `os::util::trap::init_err` (which appears to be idempotent)?

stevekuznetsov replied:
We need to `source "${OS_ROOT}/hack/lib/log/stacktrace.sh` inside of `hack/lib/util/trap.sh` to call the stacktrace functions inside of the error handler. If we also `source` the other direction, we hot-loop and segfault. 

We *could* make a dummy-function like `os::lib::log::stacktrace` and make the `source` conditional if we see that function in what is currently `set` but that seems *a)* crazy and *b)* less ideal than just `source`-ing both from your script.

However, if we are attempting to slim down the preamble in our scripts I can see how it could be useful. We could even just have a `source "${OS_ROOT}/hack/lib.sh` that `source`s the entire lib tree in such a manner.

@smarterclayton thoughts?

stevekuznetsov replied:
ping @smarterclayton

stevekuznetsov replied:
Considering `#ifndef` exists this actually doesn't sound *that* crazy.

stevekuznetsov commented:
@smarterclayton addressed comments. 

New stack trace:
```
$ hack/build-go.sh 
++ Building go targets for linux/amd64: cmd/openshift cmd/oc
# github.com/openshift/origin/pkg/cmd/util
pkg/cmd/util/env.go:16: syntax error: unexpected Int, expecting (
pkg/cmd/util/env.go:25: Env redeclared in this block
	previous declaration at pkg/cmd/util/env.go:16
[ERROR] PID 19547: hack/common.sh:311: `GOOS=${platform%/*} GOARCH=${platform##*/} go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]}"` exited with status 2.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/common.sh:311: `GOOS=${platform%/*} GOARCH=${platform##*/} go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]}"`
[INFO] 		  2: hack/common.sh:255: os::build::internal::build_binaries
[INFO] 		  3: hack/build-go.sh:36: os::build::build_binaries
[INFO]   Exiting with code 2.
[ERROR] PID 19521: hack/common.sh:253: `( os::build::internal::build_binaries "$@" )` exited with status 2.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/common.sh:253: `( os::build::internal::build_binaries "$@" )`
[INFO] 		  2: hack/build-go.sh:36: os::build::build_binaries
[INFO]   Exiting with code 2.
```

Old stack trace:
```
$ hack/build-go.sh 
++ Building go targets for linux/amd64: cmd/openshift cmd/oc
# github.com/openshift/origin/pkg/cmd/util
pkg/cmd/util/env.go:16: syntax error: unexpected Int, expecting (
pkg/cmd/util/env.go:25: Env redeclared in this block
	previous declaration at pkg/cmd/util/env.go:16
!!! Error in hack/../hack/common.sh:301
	'GOOS=${platform%/*} GOARCH=${platform##*/} go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]}"' exited with status 2
Call stack:
	1: hack/../hack/common.sh:301 os::build::build_binaries(...)
	2: hack/build-go.sh:32 main(...)
Exiting with status 1
!!! Error in hack/../hack/common.sh:253
	'( os::build::setup_env; local version_ldflags; version_ldflags=$(os::build::ldflags); local goflags; eval "goflags=(${OS_GOFLAGS:-})"; local arg; for arg in "$@";
do
    if [[ "${arg}" == -* ]]; then
        goflags+=("${arg}");
    fi;
done; os::build::export_targets "$@"; local -a nonstatics=(); local -a tests=(); for binary in "${binaries[@]}";
do
    if [[ "${binary}" =~ ".test"$ ]]; then
        tests+=($binary);
    else
        nonstatics+=($binary);
    fi;
done; local host_platform=$(os::build::host_platform); local platform; for platform in "${platforms[@]}";
do
    echo "++ Building go targets for ${platform}:" "${targets[@]}"; mkdir -p "${OS_OUTPUT_BINPATH}/${platform}"; if [[ $platform == $host_platform ]]; then
        export GOBIN="${OS_OUTPUT_BINPATH}/${platform}";
    else
        unset GOBIN;
    fi; if [[ ${#nonstatics[@]} -gt 0 ]]; then
        GOOS=${platform%/*} GOARCH=${platform##*/} go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]}"; if [[ $platform != $host_platform ]]; then
            local platform_src="/${platform//\//_}"; mv "${OS_TARGET_BIN}/${platform_src}/"* "${OS_OUTPUT_BINPATH}/${platform}/";
        fi;
    fi; for test in "${tests[@]:+${tests[@]}}";
    do
        local outfile="${OS_OUTPUT_BINPATH}/${platform}/$(basename ${test})"; GOOS=${platform%/*} GOARCH=${platform##*/} go test -c -o "${outfile}" "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "$(dirname ${test})";
    done;
done )' exited with status 1
Call stack:
	1: hack/../hack/common.sh:253 os::build::build_binaries(...)
	2: hack/build-go.sh:32 main(...)
Exiting with status 1
```

stevekuznetsov commented:
[test]me

stevekuznetsov commented:
Flake on https://github.com/openshift/origin/issues/7429

re[test]

smarterclayton commented:
Conformance flake is https://github.com/openshift/origin/issues/8491

On Mon, Apr 18, 2016 at 2:25 PM, OpenShift Bot <notifications@github.com>
wrote:

> continuous-integration/openshift-jenkins/test FAILURE (
> https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/3067/)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8527#issuecomment-211514964>
>

stevekuznetsov commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+        local bash_source="$( os::util::repository_relative_path "${BASH_SOURCE[$i]}" )"
+        if [[ -z "${preamble_finished-}" ]]; then
+            preamble_finished=true
+            os::log::error "PID ${BASHPID:-$$}: ${bash_source}:${BASH_LINENO[$i-1]}: \`${last_command}\` exited with status ${return_code}." >&2
```
@Miciah I understand `${BASHPID:-$$}` isn't technically a loss-less default, but I'm not particularly interested in making this nice for people using 12-year-old-Bash, so instead of a clear indication that each stacktrace is for a failed process, they'll just always see the parent PID.

stevekuznetsov commented:
integration flake: https://github.com/openshift/origin/issues/8573

re[test]

stevekuznetsov commented:
@Miciah were you able to take a look? If not, @eparis do you mind doing a review for this?

eparis commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+    local i
+    for (( i = stack_begin_index; i < ${#BASH_SOURCE[@]}; i++ )); do
+        local bash_source="$( os::util::repository_relative_path "${BASH_SOURCE[$i]}" )"
+        if [[ -z "${preamble_finished-}" ]]; then
```
missing `:` ?

Miciah commented on hack/common.sh:
```diff
@@ -252,7 +252,17 @@ os::build::build_static_binaries() {
+  ( os::build::internal::build_binaries "$@" )  
+}
+
+# Build binaries targets specified. Should always be run in a sub-shell so we don't leak GOBIN
```
Seems like "binaries targets" here and on line 261 should be "binary targets" instead.

Miciah commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+# Arguments:
+#  None
+# Returns:
+#  - export OS_USE_STACKTRACE
```
Here and elsewhere, you list library-internal state-tracking variables as returns, which seems like it's stretching the definition of a "return".  Does it really make sense to list such things as returns?

stevekuznetsov replied:
The Google (? it's been a while) guide asks to list external state changes that a function creates *other* than the exit code it returns with. Since this is a globally scoped variable, it seems to fit that bill. It's perhaps more useful on longer methods, though.

Miciah commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+
+# os::log::stacktrace::print prints the stacktrace and exits with the return code from the script that
+# called for a stack trace. This function will always return 0 if it is not handling the signal, and if it
+# is handling the signal, this function will always `exit`, not return, the return code it recieves as 
```
s/recieves/receives/

Miciah commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+    local errexit_set=${3:-}
+
+    if [[ "${return_code}" = "0" ]]; then
+        # we're not supposed to respond when no error has occured
```
s/occured/occurred/

Miciah commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+    # actual script code and not this handler's call
+    local stack_begin_index
+    for (( stack_begin_index = 0; stack_begin_index < ${#BASH_SOURCE[@]}; stack_begin_index++ )); do
+        if ! echo "${BASH_SOURCE[${stack_begin_index}]}" | grep -Eq 'hack/lib/(log/stacktrace|util/trap)\.sh'; then
```
Could you instead use `if [[ ${BASH_SOURCE[${stack_begin_index}]} =~ hack/lib/(log/stacktrace|util/trap)\.sh ]]`?

Miciah commented on hack/lib/log/stacktrace.sh:
```diff
@@ -0,0 +1,84 @@
+        stack_index=$(( stack_index + 1 ))
+    done
+
+    # we know we're the pivileged handler in this chain, so we can safely exit the shell without
```
s/pivileged/privileged/

Miciah commented on hack/lib/util/trap.sh:
```diff
@@ -0,0 +1,113 @@
+    local -r return_code=$?
+    local -r last_command="${BASH_COMMAND}"
+
+    if set +o | grep -q '\-o errexit'; then
```
I'd use `grep -q -e '-foo'` instead of `grep -q '\-foo'`, but I guess it's a stylistic preference.

Miciah replied:
Under what circumstances are you anticipating `os::util::trap::err_handler` could be invoked with `errexit` disabled?

stevekuznetsov replied:
There remain parts of the codebase that turn off `errexit` for short stretches.

Miciah commented on hack/lib/util/misc.sh:
```diff
@@ -0,0 +1,64 @@
+# os::util::describe_return_code describes an exit code
+#
+# Globals:
+#  - START_TIME
```
It would make sense for callers (i.e., `os::util::trap::exit_handler`) to pass this value as a parameter instead of using the global.  Making it a parameter would also avoid one potential pitfall: setting the global variable, but forgetting to export it, which would mean it would be unavailable here because `os::util::trap::exit_handler` invokes `os::util::describe_return_code` in a subshell.

stevekuznetsov replied:
By setting the value in the `'install'` function we ensure that this value is always present. This continues using the global but this is much less intensive than determining how to feed the correct value as an arg into the trap handler. I will implement this alternate approach, unless you see pitfalls to it.

Miciah commented on hack/test-go.sh:
```diff
@@ -45,8 +45,12 @@ OS_ROOT=$(dirname "${BASH_SOURCE}")/..
-os::log::install_errexit
+
+os::util::trap::init_err
+os::log::stacktrace::install
```
In other files, you are invoking `os::util::trap::init_err` and `os::log::stacktrace::install` *before* `cd "${OS_ROOT}"`.  I don't see that the discrepancy will matter with the current code, but the inconsistency kind of jumps out at me, and it could violate some subtle or future assumption.

stevekuznetsov replied:
I will fix this if we determine a better preamble process.

Miciah commented:
Nothing uses `os::util::trap::init`, nothing uses `os::util::trap::init_exit` except for `os::util::trap::init`, and nothing sets `START_TIME`.  I take it users will come in future commits?

Miciah commented on hack/test-tools.sh:
```diff
@@ -12,7 +12,12 @@ cd "${OS_ROOT}"
+source "${OS_ROOT}/hack/lib/util/trap.sh"
+source "${OS_ROOT}/hack/lib/log/stacktrace.sh"
+
+trap os::test::junit::reconcile_output EXIT
```
This looks like a fix that is not directly related to the other changes, or am I mistaken?

stevekuznetsov replied:
Yes, good eyes - I just caught this while I was fixing preambles.

Miciah commented:
Reviewed, commented on some minor things.  Sorry for the delay!

stevekuznetsov commented:
> Nothing uses os::util::trap::init

Consumers will come once exit traps are integrated into the new system and scripts want to use both.

> nothing uses os::util::trap::init_exit except for os::util::trap::init

Subtle cases come out during testing -- there's a bunch of test scripts that verify functionality of our library functions that I have in the older PR where `init_exit` is necessary but `init_err` is not, IIRC.

> and nothing sets START_TIME

This is an oversight. Currently we expect scripts to set it themselves, but I'll make it more internal.

stevekuznetsov commented:
@Miciah I've addressed your comments

stevekuznetsov commented:
I don't see a good reason to block this on the preamble work in #8743 
re[test]

stevekuznetsov commented:
Conformance flake on infrastructure, related to #8571 I think:
```
Cannot retrieve metalink for repository: epel/x86_64. Please verify its path and try again
```

re[test]

stevekuznetsov commented:
@smarterclayton we have green tests.

stevekuznetsov commented:
@smarterclayton rebased, pending tests

smarterclayton commented:
LGTM [merge] but I hope it doesn't break me....

smarterclayton commented:
Failed the networking tests.

smarterclayton commented:
Removing merge.

stevekuznetsov commented on hack/lib/util/trap.sh:
```diff
@@ -0,0 +1,99 @@
+#  - returns original return code, allows privileged handler to exit if necessary
+function os::util::trap::err_handler() {
+    local -r return_code=$?
+    local -r last_command="${BASH_COMMAND}"
```
Note to self:
```
-    local -r last_command="${BASH_COMMAND}"
+    local -r last_command_literal="${BASH_COMMAND}"
+    local -r last_command="$( eval "echo ${last_command_literal}" )"
```

stevekuznetsov commented:
Rebased on #8853 and added an `eval` ... new output:
```
$ hack/build-go.sh 
++ Building go targets for linux/amd64: cmd/openshift cmd/oc
# github.com/openshift/origin/pkg/cmd/cli/cmd
_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/logout.go:48: syntax error: unexpected ut, expecting (
[ERROR] PID 16034: hack/common.sh:317: `GOOS=${platform%/*} GOARCH=${platform##*/} go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]}"` exited with status 2.
[ERROR] Command with variables substituted is: 
[ERROR] 	GOOS=linux GOARCH=amd64 go install  -ldflags -X github.com/openshift/origin/pkg/version.majorFromGit=1 -X github.com/openshift/origin/pkg/version.minorFromGit=3+ -X github.com/openshift/origin/pkg/version.versionFromGit=v1.3.0-alpha.1-245-g7574e4e-dirty -X github.com/openshift/origin/pkg/version.commitFromGit=7574e4e -X k8s.io/kubernetes/pkg/version.gitCommit=7574e4e -X k8s.io/kubernetes/pkg/version.gitVersion=v1.3.0-alpha.1-331-g0522e63 github.com/openshift/origin/cmd/openshift github.com/openshift/origin/cmd/oc
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/common.sh:317: `GOOS=${platform%/*} GOARCH=${platform##*/} go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]}"`
[INFO] 		  2: hack/common.sh:261: os::build::internal::build_binaries
[INFO] 		  3: hack/build-go.sh:31: os::build::build_binaries
[INFO]   Exiting with code 2.
[ERROR] PID 16006: hack/common.sh:259: `( os::build::internal::build_binaries "$@" )` exited with status 2.
[ERROR] Command with variables substituted is: 
[ERROR] 	( os::build::internal::build_binaries  )
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/common.sh:259: `( os::build::internal::build_binaries "$@" )`
[INFO] 		  2: hack/build-go.sh:31: os::build::build_binaries
[INFO]   Exiting with code 2.
```

The largest of the caveats (all of which are in the commit message) is that we can't correctly evaluate positional parameters `$1, $2, $@, $*` ... but I contend that "good" functions will always translate positional arguments into readonly, named args at the start for readability, etc, so hopefully this won't be a huge issue.

smarterclayton commented:
That's a shame we don't get positional.

On Mon, Jun 13, 2016 at 10:20 AM, OpenShift Bot <notifications@github.com>
wrote:

> continuous-integration/openshift-jenkins/test FAILURE (
> https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/4762/)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8527#issuecomment-225595295>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p7lPUQ7hBny_45rdYfqakQrBgL5Xks5qLWc_gaJpZM4IIccN>
> .
>

stevekuznetsov commented:
> That's a shame we don't get positional.

Shame, but in most cases where we're not using arrays of positionals (`$@`) we should really bring in the positionals and give them names right away so we can fail early and fast as well as making the code easier to read. So ultimately with well-factored libraries this isn't a huge issue.

stevekuznetsov commented:
AWS Infra flake:
```
The instance never became "ready" in AWS.
```

re[test]

stevekuznetsov commented:
re[test]

stevekuznetsov commented:
@smarterclayton this is ready
On Jun 15, 2016 5:50 PM, "OpenShift Bot" <notifications@github.com> wrote:

> continuous-integration/openshift-jenkins/test SUCCESS (
> https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/4941/)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8527#issuecomment-226330847>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG_ScmgzbUOB93vrblkzslv4RJoVrbNtks5qMHOtgaJpZM4IIccN>
> .
>

stevekuznetsov commented:
Well, there was a green run before. Not sure what kicked this one.

e2e flake:
```
Running test/end-to-end/core.sh:351: executing 'docker pull gcr.io/google_containers/pause' expecting success...
FAILURE after 0.309s: test/end-to-end/core.sh:351: executing 'docker pull gcr.io/google_containers/pause' expecting success: the command returned the wrong error code
Standard output from the command:
Using default tag: latest
Trying to pull repository gcr.io/google_containers/pause ... 

Standard error from the command:
name invalid: Requested repository does not match bearer token resource: google_containers/pause
```

test-cmd flake:
```
FAILURE after 0.214s: test/cmd/router.sh:32: executing 'oadm router --dry-run --host-network=false --stats-port=1937 -o yaml' expecting failure and text 'hostPort: 1937': the output content test failed
Standard output from the command:
...
          - containerPort: 1937
            hostPort: 1937
...

Standard error from the command:
error: router could not be created; service account "router" is not allowed to access host ports on nodes, grant access with oadm policy add-scc-to-user hostnetwork -z router
```

re[test]

stevekuznetsov commented:
Looks like a new flake in conformance and integration:
```
curl: (7) Failed connect to copr.fedorainfracloud.org:443; Connection refused
```
https://github.com/openshift/origin/issues/9379

re[test]

stevekuznetsov commented:
Extended failures: 
 - `yum` metalink https://github.com/openshift/origin/issues/8571
 - #9364 

`test-cmd`:
```
FAILURE after 0.415s: test/cmd/admin.sh:308: executing 'oadm registry --dry-run' expecting failure and text 'does not exist': the output content test failed
There was no output from the command.
Standard error from the command:
error: error getting client: Get https://127.0.0.1:28443/api: dial tcp 127.0.0.1:28443: getsockopt: connection refused
```
@smarterclayton you seen this before? ^^

re[test]

stevekuznetsov commented:
@mfojtik @smarterclayton this looks like a rebase issue?
```
I0616 16:05:52.554723   19712 meta.go:46] Calling Accessor on non-internal object: *extensions.DaemonSetList
panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *extensions.DaemonSet

goroutine 5893 [running]:
panic(0x35dca60, 0xc82c065e80)
	/usr/lib/golang/src/runtime/panic.go:481 +0x3e6
k8s.io/kubernetes/pkg/controller/daemon.NewDaemonSetsController.func5(0x3611500, 0xc8274fe840)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/daemon/controller.go:172 +0x18e
k8s.io/kubernetes/pkg/controller/framework.ResourceEventHandlerFuncs.OnDelete(0xc8250a00a0, 0xc8250a00b0, 0xc8250a00c0, 0x3611500, 0xc8274fe840)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:184 +0x3a
k8s.io/kubernetes/pkg/controller/framework.(*ResourceEventHandlerFuncs).OnDelete(0xc82785aea0, 0x3611500, 0xc8274fe840)
	<autogenerated>:25 +0xb5
k8s.io/kubernetes/pkg/controller/framework.NewInformer.func1(0x36116c0, 0xc8274feca0, 0x0, 0x0)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:254 +0x56d
k8s.io/kubernetes/pkg/controller/framework.(*Controller).processLoop(0xc8271d90a0)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:128 +0x6f
k8s.io/kubernetes/pkg/controller/framework.(*Controller).(k8s.io/kubernetes/pkg/controller/framework.processLoop)-fm()
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:103 +0x20
k8s.io/kubernetes/pkg/util/wait.JitterUntil.func1(0xc8280fff78)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:66 +0x4b
k8s.io/kubernetes/pkg/util/wait.JitterUntil(0xc8280fff78, 0x3b9aca00, 0x0, 0xc82005ae40)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:67 +0x68
k8s.io/kubernetes/pkg/util/wait.Until(0xc8280fff78, 0x3b9aca00, 0xc82005ae40)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:47 +0x3e
k8s.io/kubernetes/pkg/controller/framework.(*Controller).Run(0xc8271d90a0, 0xc82005ae40)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:103 +0x1c2
created by k8s.io/kubernetes/pkg/controller/daemon.(*DaemonSetsController).Run
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/daemon/controller.go:224 +0xa6
```

mfojtik commented:
it seems so, label it as post rebase, I will have a look tomorrow.

Sent from my phone

On 16 Jun 2016, at 23:42, Steve Kuznetsov <notifications@github.com> wrote:

@mfojtik <https://github.com/mfojtik> @smarterclayton
<https://github.com/smarterclayton> this looks like a rebase issue?

I0616 16:05:52.554723   19712 meta.go:46] Calling Accessor on
non-internal object: *extensions.DaemonSetList
panic: interface conversion: interface {} is
cache.DeletedFinalStateUnknown, not *extensions.DaemonSet

goroutine 5893 [running]:
panic(0x35dca60, 0xc82c065e80)
    /usr/lib/golang/src/runtime/panic.go:481
+0x3e6k8s.io/kubernetes/pkg/controller/daemon.NewDaemonSetsController.func5(0x3611500,
0xc8274fe840)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/daemon/controller.go:172
+0x18ek8s.io/kubernetes/pkg/controller/framework.ResourceEventHandlerFuncs.OnDelete(0xc8250a00a0,
0xc8250a00b0, 0xc8250a00c0, 0x3611500, 0xc8274fe840)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:184
+0x3ak8s.io/kubernetes/pkg/controller/framework.(*ResourceEventHandlerFuncs).OnDelete(0xc82785aea0,
0x3611500, 0xc8274fe840)
    <autogenerated>:25
+0xb5k8s.io/kubernetes/pkg/controller/framework.NewInformer.func1(0x36116c0,
0xc8274feca0, 0x0, 0x0)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:254
+0x56dk8s.io/kubernetes/pkg/controller/framework.(*Controller).processLoop(0xc8271d90a0)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:128
+0x6fk8s.io/kubernetes/pkg/controller/framework.(*Controller).(k8s.io/kubernetes/pkg/controller/framework.processLoop)-fm()
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:103
+0x20k8s.io/kubernetes/pkg/util/wait.JitterUntil.func1(0xc8280fff78)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:66
+0x4bk8s.io/kubernetes/pkg/util/wait.JitterUntil(0xc8280fff78,
0x3b9aca00, 0x0, 0xc82005ae40)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:67
+0x68k8s.io/kubernetes/pkg/util/wait.Until(0xc8280fff78, 0x3b9aca00,
0xc82005ae40)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:47
+0x3ek8s.io/kubernetes/pkg/controller/framework.(*Controller).Run(0xc8271d90a0,
0xc82005ae40)
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:103
+0x1c2
created by k8s.io/kubernetes/pkg/controller/daemon.(*DaemonSetsController).Run
    /data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/daemon/controller.go:224
+0xa6

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/8527#issuecomment-226622396>, or mute
the thread
<https://github.com/notifications/unsubscribe/AACsaMRifV5pIprQ26l1PfW3r2PNOulNks5qMcMvgaJpZM4IIccN>
.

stevekuznetsov commented:
re[test]

stevekuznetsov commented:
flake 2x https://github.com/openshift/origin/issues/8571

re[test]

stevekuznetsov commented:
Hit https://github.com/openshift/origin/issues/8571 again

re[test]

stevekuznetsov commented:
hit https://github.com/openshift/origin/issues/9457
re[test]

stevekuznetsov commented:
hit https://github.com/openshift/origin/issues/5448

re[test]

smarterclayton commented:
LGTM [merge]

stevekuznetsov commented:
hit https://github.com/openshift/origin/issues/5448
@smarterclayton please re-tag

stevekuznetsov commented:
I never thought the day would come.
:fireworks: :confetti_ball: :tada: :fireworks: :confetti_ball: :tada: :fireworks: :confetti_ball: :tada: :fireworks: :confetti_ball: :tada:

mfojtik commented:
@stevekuznetsov I know that feeling

rhcarvalho commented:
Who knows, maybe one day it will become https://github.com/niieani/bash-oo-framework#error-handling-with-exceptions-and-throw...

stevekuznetsov commented:
@rhcarvalho good lord ...
